lib/resque/plugins/retry.rb

```diff
@@ -564,11 +580,24 @@ def give_up_callback(method=nil, &block)
       # @api private
       def run_give_up_callbacks(exception, *args)
         give_up_callbacks.each do |callback|
-          if callback.is_a? Symbol
-            send(callback, exception, *args)
-          elsif callback.respond_to?(:call)
-            callback.call(exception, *args)
-          end
+          call_symbol_or_block(callback, exception, *args)
+        end
+      end
+
+      # Helper to call functions that may be passed as Symbols or Procs. If
+      # a symbol, it is assumed to refer to a method that is already defined
+      # on this class.
+      #
+      # @param [Symbol|Proc] method
+      # @param [Object...] *args
+      # @return [Object]
+      #
+      # @api private
+      def call_symbol_or_block(method, *args)
+        if method.is_a?(Symbol)
+          send(method, *args)
+        elsif method.respond_to?(:call)
+          instance_exec(*args, &method)
```

Comment:
There is an effect that now `instance_exec` instead of `method.call` is used for the `try_again_callbacks` and `give_up_callbacks` too. I'm not sure whether there is any undesirable side effect in those cases.